### ISSUE TITLE
Source text not aligned to middle while loading a beatmap.

### ISSUE THREAD

**Issue by dscnnctd** [ppy/osu#4506]
**Issue:** I don’t think I should be opening this issue but isn’t the source text supposed to be aligned automatically in the middle if the beatmap has a source?
**Version:** _2019.321.0_
**Image:**
![GitHawk Upload by dscnnctd](https://i.imgur.com/Qf8aDvW.jpg)

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by Joehuu** [comment]
This was a closed issue #2414.

**Comment by Joehuu** [comment]
This is not a regression. In the referenced issue, you said this layout was correct.

**Comment by peppy** [comment]
My comment in the last issues was made under the presumption that the issue was fixed, not that the reported behaviour was correct.

**Comment by Joehuu** [comment]
Need a design for this then as my past proposals were rejected.

**Comment by peppy** [comment]
It should be centre aligned

**Comment by Joehuu** [comment]
Wasn't the past PR I made right then? https://github.com/ppy/osu/pull/2795#issuecomment-397134066

![](https://puu.sh/D3wk5/ba48e52d8c.png)

**Comment by peppy** [comment]
Yes, that is wrong. the full *block* should be centered.

**Comment by aemino** [comment]
![image](https://user-images.githubusercontent.com/13167687/68738906-640f9000-059c-11ea-84cb-45a8daf284f8.png)

Should it look like this?

**Comment by dscnnctd** [comment]
> ![image](https://user-images.githubusercontent.com/13167687/68738906-640f9000-059c-11ea-84cb-45a8daf284f8.png)
> 
> Should it look like this?

@peppy approves this proposal but he wants some additional changes shown in the image below:
![7EA77B8B-B1BB-4435-BE66-F15C610E654B](https://user-images.githubusercontent.com/48611569/68739315-24ea3a80-05ae-11ea-9876-04cec3ca07fd.jpeg)

**Comment by aemino** [comment]
I'm new to the osu codebase so I'm not completely sure how I would go about that, should I make the `MetadataLine`s be `TextFlowContainer`s?

**Comment by dscnnctd** [comment]
> I'm new to the osu codebase so I'm not completely sure how I would go about that, should I make the `MetadataLine`s be `TextFlowContainer`s?

![7188177E-603E-48E1-B490-B9F37F6042AF](https://user-images.githubusercontent.com/48611569/68817968-02592f80-0649-11ea-8e74-6a2fe3b7519e.jpeg)

**Comment by aemino** [comment]
Are there any other options for accomplishing wrapping then?

(btw I'm in the osu!dev server so I can see your comments there :P)

**Comment by bdach** [comment]
I mean a `TextFlowContainer` with a max width is probably correct and desired here in the long term, but right now it won't do the proper thing for long words (especially CJK chars). I'm not sure if it would be better to break them or truncate them here. Both won't look too great, breaking would probably be better?

**Comment by Tudi20** [comment]
Is this still relevant?
![Current](https://user-images.githubusercontent.com/13798839/79781579-f717d780-833d-11ea-8966-026811fac301.png)

**Comment by Joehuu** [comment]
![image](https://user-images.githubusercontent.com/35318437/79781742-cfe7e880-82f2-11ea-8a2e-f69931e0817b.png)

Yes.

**Comment by Luckyfox1111** [comment]
Can I work on this ?

**Comment by bdach** [comment]
I'm not necessarily sure whether this actually *is* a good first issue due to all the reasons listed above, but I suppose you could take a look and see if you can do anything about this. Please do read the [contributing guidelines](https://github.com/ppy/osu/blob/master/CONTRIBUTING.md) and the README before starting, though.
